design/use_cases/keep_out_zones_use_case.md

```diff
@@ -0,0 +1,15 @@
+# Keep Out Zones
+As a Robot user I want to be able to designate keep-out zones or areas on a map so that my robot will go around those areas instead of through them 
+
+## More details
+- Why is this needed?
+   - This is needed for indoor/outdoor robot navigation in areas where there may be safety issues or hazards
+   - Example: a courier robot in a logistics warehouse may be required to avoid areas where the forklift is unloading pallets from trucks
+
+- What is the expected user interaction?
+   - The user should be able to specify a keep out zones on a map through a GUI 
+   - The keep out zones should also be able to be specified programmatically through an API or config file
+
```

Comment:
How about: - The user can define keep out areas by marking the floor with tape.

Reply:
I recommend keeping the use case at a high level to avoid specific implementation decisions. In this case, for example, the intent is to identify keep-out zones. That can be accomplished in multiple ways, which don't have to be specified here. So, we could drop the "on a map" part.

Reply:
@orduno - I'm not sure about that one, I don't know that anyone really expects to be able to do that. It would be great however if we could support it in some way, but probably an enhancement / nice to have feature.

Reply:
@mjeronimo - Is it an implementation decision to say the user expects to be able to use a GUI to designate a keep-out-zone?

Reply:
@mkhansen-intel Yes. As Carlos has pointed out a keep-out zone could be defined another way, such as marking the floor with tape. The intention is that the Robot User is able to define the keep-out zones. The requirement is the same, but different systems could satisfy this requirement in different ways. We may decide to use a GUI to designate the keep-out zone, and that could even be included as a suggestion, but it isn't itself the use case/requirement, IMO.

Reply:
@mjeronimo - OK, I'll remove the GUI, Map, API parts from the UC and keep it more abstract for now, we can add more details later in the requirements level